src/Npgsql/CrateDbDatabaseInfo.cs

```diff
@@ -0,0 +1,235 @@
+        public override bool SupportsUnlisten => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports range types.
+        /// </summary>
+        public override bool SupportsRangeTypes => false;
+        /// <summary>
+        /// Returns a boolean value that signals if the current database supports advisory lock functions.
+        /// </summary>
+        public override bool SupportsAdvisoryLocks => false;
+        /// <summary>
+        /// Reports whether the backend uses the newer integer timestamp representation. 
+        /// CrateDB used floating point timestamps until version &lt;=3.0 and switched to integer datetimes with version &gt;= 3.1.
+        /// </summary>
+        public override bool HasIntegerDateTimes { get; protected set; } = true;
+
+        static IEnumerable<NpgsqlTypeMapping> CrateDbSpecificTypeMappings()
+        {
+            // Map CrateDB varchar type to the Npgsql TextHandler.
+            yield return new NpgsqlTypeMappingBuilder
+            {
+                PgTypeName = "character varying",
+                NpgsqlDbType = NpgsqlDbType.Varchar,
+                DbTypes = new[] { DbType.String, DbType.StringFixedLength, DbType.AnsiString, DbType.AnsiStringFixedLength },
+                ClrTypes = new[] { typeof(string), typeof(char[]), typeof(char) },
+                InferredDbType = DbType.String,
+                TypeHandlerFactory = new TextHandlerFactory()
+            }
+            .Build();
+
+            // Map CrateDB timestampz type to the CrateDbTimestampHandler.
+            yield return new NpgsqlTypeMappingBuilder
+            {
+                PgTypeName = "timestamp with time zone",
+                NpgsqlDbType = NpgsqlDbType.TimestampTz,
+                DbTypes = new[] { DbType.DateTime },
+                ClrTypes = new[] { typeof(DateTime) },
+                TypeHandlerFactory = new TimestampHandlerFactory()
+            }
+            .Build();
+        }
+
+        static bool IsPgTypeSupportedByCrateDb(string pgTypeName)
+        {
+            var internalName = PgTypeNameToInternalName.ContainsKey(pgTypeName) ? PgTypeNameToInternalName[pgTypeName] : pgTypeName;
+            return CrateDbBaseTypes.ContainsKey(internalName) || CrateDbArrayTypes.ContainsKey(internalName);
+        }
+    }
+
+    /// <summary>
+    /// Represents a base type supported by CrateDB.
+    /// </summary>
+    public class CrateDbBaseType : PostgresBaseType
+    {
+        /// <summary>
+        /// Creates an instance of the CrateDbBaseType class.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="oid"></param>
+        public CrateDbBaseType(string name, uint oid)
+            : base("pg_catalog", name, oid)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Represents a array type supported by CrateDB.
+    /// </summary>
+    public class CrateDbArrayType : PostgresArrayType
```

Comment:
Are these base classes in separate files? If so, I would break these out class-per-file style. If not, please disregard.

Reply:
yeah you are right i moved them to separate files